command_test.go

```diff
@@ -0,0 +1,126 @@
+// Copyright 2016 Keybase, Inc. All rights reserved. Use of
+// this source code is governed by the included BSD license.
+
+package updater
+
+import (
+	"os"
+	"reflect"
+	"testing"
+	"time"
+
+	"github.com/keybase/go-logging"
+)
+
+func TestEmptyRunCommand(t *testing.T) {
+	log := logging.Logger{Module: "test"}
+	_, err := RunCommand("", nil, time.Second, log)
+	t.Logf("Error: %s", err)
+	if err == nil {
+		t.Fatal("Should have errored")
+	}
+}
+
+func TestInvalidRunCommand(t *testing.T) {
+	log := logging.Logger{Module: "test"}
+	_, err := RunCommand("invalidexecutable", nil, time.Second, log)
+	t.Logf("Error: %s", err)
+	if err == nil {
+		t.Fatal("Should have errored")
+	}
+}
+
+func TestRunCommandEcho(t *testing.T) {
+	log := logging.Logger{Module: "test"}
+	out, err := RunCommand("echo", []string{"arg1", "arg2"}, time.Second, log)
+	if err != nil {
+		t.Fatal(err)
+	}
+	expected := "arg1 arg2\n"
+	if out != expected {
+		t.Errorf("Unexpected output: %q != %q", out, expected)
+	}
+}
+
+func TestRunCommandTimeout(t *testing.T) {
+	log := logging.Logger{Module: "test"}
+	_, err := RunCommand("sleep", []string{"10"}, time.Second, log)
+	if err == nil {
+		t.Fatal("Expected timeout error")
+	}
+	if err.Error() != "Error running command: signal: killed" {
+		t.Errorf("Expected signal killed error, got %#v", err)
+	}
+}
+
+type testObj struct {
+	StringVar string `json:"stringVar"`
+	NumberVar int    `json:"numberVar"`
+	BoolVar   bool   `json:"boolVar"`
+}
+
+const testJSON = `{
```

Comment:
maybe this should have at least 1 level of nested object, like sub: {"hello": "world"}